In [26]:
from keras.preprocessing.text import one_hot

In [4]:
sent= ['i am a girl',
       'i love anime',
       'the milk is hot',
       'the coffee is bitter',
       'apple is red in colour',
       'ai is evovling everyday',
       'a black hole is very beautiful',
       'organic chemistry is all about carbon',
       'cats are cute',
       'luffy is one of the best main lead']

In [6]:
sent

['i am a girl',
 'i love anime',
 'the milk is hot',
 'the coffee is bitter',
 'apple is red in colour',
 'ai is evovling everyday',
 'a black hole is very beautiful',
 'organic chemistry is all about carbon',
 'cats are cute',
 'luffy is one of the best main lead']

In [19]:
vocab= 7000

In [20]:
on_hot= [one_hot(word ,vocab) for word in sent]

In [22]:
print(on_hot)

[[6546, 3437, 5738, 1611], [6546, 3019, 5974], [5477, 4836, 639, 6978], [5477, 6762, 639, 6172], [2015, 639, 6814, 4806, 4940], [216, 639, 6353, 6023], [5738, 1210, 6475, 639, 5178, 437], [5047, 3460, 639, 5075, 1639, 2827], [6757, 1649, 3464], [3065, 639, 6375, 2146, 5477, 5188, 300, 4333]]


In [31]:
from keras.models import Sequential
from keras.layers import Embedding
from tensorflow.keras.preprocessing.sequence import pad_sequences
import numpy as np

In [42]:
sent_length= 8
emb= pad_sequences(on_hot, padding='post', maxlen= sent_length)

In [43]:
emb

array([[6546, 3437, 5738, 1611,    0,    0,    0,    0],
       [6546, 3019, 5974,    0,    0,    0,    0,    0],
       [5477, 4836,  639, 6978,    0,    0,    0,    0],
       [5477, 6762,  639, 6172,    0,    0,    0,    0],
       [2015,  639, 6814, 4806, 4940,    0,    0,    0],
       [ 216,  639, 6353, 6023,    0,    0,    0,    0],
       [5738, 1210, 6475,  639, 5178,  437,    0,    0],
       [5047, 3460,  639, 5075, 1639, 2827,    0,    0],
       [6757, 1649, 3464,    0,    0,    0,    0,    0],
       [3065,  639, 6375, 2146, 5477, 5188,  300, 4333]], dtype=int32)

In [44]:
dim= 10
model= Sequential()
model.add(Embedding(vocab, 10, input_length= sent_length))
model.compile('adam', 'mse')

In [45]:
model.summary()

Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 embedding_1 (Embedding)     (None, 8, 10)             70000     
                                                                 
Total params: 70,000
Trainable params: 70,000
Non-trainable params: 0
_________________________________________________________________


In [46]:
print(model.predict(emb))

1/1 [==============================] - 0s 42ms/step
[[[ 1.04896910e-02  2.34019794e-02 -3.83937955e-02  1.06878765e-02
    3.37150209e-02  4.39604633e-02 -3.94153967e-02  2.63463520e-02
   -1.36795528e-02  2.62516737e-03]
  [-1.29825473e-02 -1.10447779e-02  2.41856314e-02  4.76833470e-02
    2.47554444e-02 -1.36876479e-02 -1.95396300e-02 -4.79974411e-02
   -3.21910381e-02 -2.90167332e-03]
  [-8.71169567e-03  3.29580568e-02  2.58002616e-02  3.86929028e-02
   -3.68338712e-02  2.09208764e-02  1.02859735e-02  2.50026844e-02
   -2.68104076e-02 -2.94670463e-05]
  [ 8.54548067e-03 -9.57620144e-03  1.28866769e-02  4.65941541e-02
    6.20671362e-03  7.70078972e-03 -1.05801597e-02 -4.11707647e-02
   -1.80612430e-02  1.13382563e-02]
  [ 3.01964991e-02 -3.39335203e-02  3.48622538e-02  4.98910062e-02
    6.87191635e-03  1.82911046e-02 -4.76382039e-02 -4.60679196e-02
   -4.79558818e-02  4.03852388e-03]
  [ 3.01964991e-02 -3.39335203e-02  3.48622538e-02  4.98910062e-02
    6.87191635e-03  1.82911046e

In [50]:
on_hot[1]

[6546, 3019, 5974]

In [49]:
emb[1]

array([6546, 3019, 5974,    0,    0,    0,    0,    0], dtype=int32)

In [48]:
print(model.predict(emb)[1])

1/1 [==============================] - 0s 31ms/step
[[ 0.01048969  0.02340198 -0.0383938   0.01068788  0.03371502  0.04396046
  -0.0394154   0.02634635 -0.01367955  0.00262517]
 [ 0.04313239  0.03829484 -0.04303659 -0.03771988  0.00404385  0.0083349
   0.04955442  0.0266319   0.04574156  0.03974874]
 [ 0.04994695  0.01082141 -0.04193278 -0.03221568  0.02962967 -0.01760054
   0.00758597 -0.04772569 -0.01059479  0.02605755]
 [ 0.0301965  -0.03393352  0.03486225  0.04989101  0.00687192  0.0182911
  -0.0476382  -0.04606792 -0.04795588  0.00403852]
 [ 0.0301965  -0.03393352  0.03486225  0.04989101  0.00687192  0.0182911
  -0.0476382  -0.04606792 -0.04795588  0.00403852]
 [ 0.0301965  -0.03393352  0.03486225  0.04989101  0.00687192  0.0182911
  -0.0476382  -0.04606792 -0.04795588  0.00403852]
 [ 0.0301965  -0.03393352  0.03486225  0.04989101  0.00687192  0.0182911
  -0.0476382  -0.04606792 -0.04795588  0.00403852]
 [ 0.0301965  -0.03393352  0.03486225  0.04989101  0.00687192  0.0182911
  -0.

10 dimension have been assigned for the first word whose value is 6546
similar way 10 dimension have been assigned to other values